lighthouse-core/computed/metrics/cumulative-long-queuing-delay.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('../computed-artifact.js');
+const ComputedMetric = require('./metric.js');
+const LHError = require('../../lib/lh-error.js');
+const TracingProcessor = require('../../lib/traces/tracing-processor.js');
+const LanternCumulativeLongQueuingDelay = require('./lantern-cumulative-long-queuing-delay.js');
+const TimetoInteractive = require('./interactive.js');
+
+/**
+ * @fileoverview This audit determines the largest 90 percentile EQT value of all 5s windows between
+ *    FMP and the end of the trace.
+ * @see https://docs.google.com/document/d/1b9slyaB9yho91YTOkAQfpCdULFkZM9LqsipcX3t7He8/preview
+ */
+class CumulativeLongQueuingDelay extends ComputedMetric {
+  /**
+   * @return {number}
+   */
+  static get LONG_QUEUING_DELAY_THRESHOLD() {
+    return 50;
+  }
+  /**
+   * @param {Array<{start: number, end: number, duration: number}>} topLevelEvents
+   * @param {number} fcpTimeInMs
+   * @param {number} interactiveTimeMs
+   * @return {number}
+   */
+  static calculateSumOfLongQueuingDelay(topLevelEvents, fcpTimeInMs, interactiveTimeMs) {
+    const threshold = CumulativeLongQueuingDelay.LONG_QUEUING_DELAY_THRESHOLD;
+    const longQueuingDelayRegions = [];
+    // [              250ms Task                   ]
+    // |  Long Queuing Delay Region  |   Last 50ms |
+    //           200 ms
+    for (const event of topLevelEvents) {
+      if (event.duration < threshold) continue;
+      longQueuingDelayRegions.push({
+        start: event.start,
```

Comment:
conceptually would we want `{start: event.start + threshold, end: event.end}` since that's the extra part we're considering delay? like the first 50ms are free but they didn't respond by then so now we're delaying

I know it wouldn't affect the metric at all, so probably moot haha

Reply:
Ah, I see. In other parts of the codebase in this situation we clip the `start/end` times so that `start` in your example case would be FCP and both accounting methods would give us 0.

Apologies for the noise I get it now!